staging/src/k8s.io/apiserver/pkg/server/options/recommended.go

```diff
@@ -57,9 +60,14 @@ func (o *RecommendedOptions) AddFlags(fs *pflag.FlagSet) {
 	o.Audit.AddFlags(fs)
 	o.Features.AddFlags(fs)
 	o.CoreAPI.AddFlags(fs)
+	o.Admission.AddFlags(fs)
 }
 
-func (o *RecommendedOptions) ApplyTo(config *server.RecommendedConfig) error {
+func (o *RecommendedOptions) ApplyTo(
+	config *server.RecommendedConfig,
+	scheme *runtime.Scheme,
+	pluginInitializers ...admission.PluginInitializer,
```

Comment:
I would turn this implicit slice into an explicit one. Otherwise, we cannot easily extend the signature anymore.

After having said that, what about moving the initializers into the AdmissionOptions ?

Reply:
just to double check: are the default initializers of k8s.io/apiserver applied even if this list is empty? They should.

Reply:
or if we move this to the AdmissionOptions, the default there should include the default initializers.

Reply:
>what about moving the initializers into the AdmissionOptions ?

The RecommendedOptions contains kinds of options. Include initializers sounds weired.

>just to double check: are the default initializers of k8s.io/apiserver applied even if this list is empty? They should.

Even if pluginInitializers list here is empty, default initializer is applied in AdmissionOptions.ApplyTo

Reply:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/server/options/admission.go#L117

Reply:
ok, then I would say to just turn it into an explicit slice here and add godoc above that it is optional and can be nil (because it's only need for additional initializers).

The scheme should never be nil, even if the admission options are not applied (we need some day need it for other things). We should check that and fail otherwise.

Reply:
make sense